src/basilisp/lang/interfaces.py

```diff
@@ -374,6 +374,18 @@ def pop(self: Self) -> Self:
         raise NotImplementedError()
 
 
+class IReduce:
+    @abstractmethod
+    def reduce(self, f, init):
+        raise NotImplementedError()
+
+
+class IReduceKV:
+    @abstractmethod
+    def reduce_kv(self, f, init):
+        raise NotImplementedError()
```

Comment:
All interfaces are pulled into the documentation, so each needs a blurb and should reference the respective functions that depend on them. Other interfaces docstrings can serve as examples.

Reply:
Thanks, I wanted to wait until I was sure you wouldn't reject me outright lol. I've added documentation.